clio-server/src/main/scala/org/broadinstitute/clio/server/service/ServerService.scala

```diff
@@ -57,27 +55,30 @@ class ServerService private (
   ): Future[Unit] = {
     val name = index.indexName
 
-    searchDAO.getMostRecentDocument.flatMap { mostRecent =>
+    searchDAO.getMostRecentDocument(index).flatMap { mostRecent =>
       val msg = s"Recovering all upserts from index $name"
-      val latestUpsert = mostRecent.map(getUpsertId)
+      val latestUpsert = mostRecent.map(ElasticsearchIndex.getUpsertId)
 
       logger.info(latestUpsert.fold(msg)(id => s"$msg since ${id.id}"))
 
       persistenceDAO
         .getAllSince(latestUpsert)
-        .batch(ServerService.RecoveryMaxBulkSize, json => Map(getEntityId(json) -> json)) {
-          (idMap, json) =>
-            val id = getEntityId(json)
-            val newJson = idMap.get(id) match {
-              case Some(oldJson) => {
-                logger.debug(
-                  s"Merging upserts ${getUpsertId(oldJson)} and ${getUpsertId(json)} for id $id"
-                )
-                oldJson.deepMerge(json)
-              }
-              case None => json
+        .batch(
+          ServerService.RecoveryMaxBulkSize,
+          json => Map(ElasticsearchIndex.getEntityId(json) -> json)
+        ) { (idMap, json) =>
+          val id = ElasticsearchIndex.getEntityId(json)
+          val newJson = idMap.get(id) match {
+            case Some(oldJson) => {
+              logger.debug(
+                s"Merging upserts ${ElasticsearchIndex.getUpsertId(oldJson)} and ${ElasticsearchIndex
+                  .getUpsertId(json)} for id $id"
```

Comment:
Curse ye, scalafmt... I think `logger.debug` can be passed a block, if you want to clean up a bit:
```scala
logger.debug {
  val oldId = ElasticsearchIndex.getUpsertId(oldJson)
  val newId = ...
  s"..."
}
```